runtime/security.go

```diff
@@ -197,8 +201,14 @@ func (p *securityEngine) resolveSecurity(instanceID, environment string, vars ma
 		return ResolvedSecurityOpen, nil
 	}
 
+	rules, err := p.expandRules(context.Background(), instanceID, claims)
+	if err != nil {
+		return nil, fmt.Errorf("failed to expand security rules: %w", err)
+	}
+	claims.AdditionalRules = rules
+
 	// Combine rules with any contained in the resource itself
-	rules := p.resolveRules(claims, r)
+	rules = p.resolveRules(claims, r)
```

Comment:
Feels a little dirty to overwrite the claims in-place, which is an unexpected side effect that maybe could lead to some edge cases in upstream logic. Could this be handled in a functional way?